Meta/Lagom/Tools/CodeGenerators/LibUnicode/GenerateUnicodeLocale.cpp

```diff
@@ -745,26 +867,25 @@ Optional<StringView> resolve_@enum_snake@_alias(StringView const& @enum_snake@)
     generator.append(R"~~~(
 void resolve_complex_language_aliases(Unicode::LanguageID& language_id)
 {
-    static auto const& complex_alias_map = ensure_complex_alias_map();
-
-    for (auto const& map : complex_alias_map) {
+    for (auto const& map : s_complex_alias) {
         if ((map.key.language != language_id.language) && (map.key.language != "und"sv))
             continue;
-        if (map.key.script.has_value() && (map.key.script != language_id.script))
+        if (!map.key.script.is_empty() && (map.key.script != language_id.script))
             continue;
-        if (map.key.region.has_value() && (map.key.region != language_id.region))
+        if (!map.key.region.is_empty() && (map.key.region != language_id.region))
             continue;
-        if (!map.key.variants.is_empty() && (map.key.variants != language_id.variants))
+        if (!map.key.matches_variants(language_id.variants))
             continue;
 
-        auto alias = map.alias;
+        auto alias = map.alias.to_unicode_language_id();
+
         if (alias.language == "und"sv)
             alias.language = move(language_id.language);
-        if (!map.key.script.has_value() && !alias.script.has_value())
+        if (map.key.script.is_empty() && !alias.script.has_value())
             alias.script = move(language_id.script);
-        if (!map.key.region.has_value() && !alias.region.has_value())
+        if (map.key.region.is_empty() && !alias.region.has_value())
             alias.region = move(language_id.region);
-        if (map.key.variants.is_empty() && alias.variants.is_empty())
+        if ((map.key.variants_size == 0) && alias.variants.is_empty())
```

Comment:
Redundant parenthesis here?
```suggestion
        if (map.key.variants_size == 0 && alias.variants.is_empty())
```

Reply:
eh, explicit parentheses everywhere is a habit I've gotten into after one too many weird bugs working on real old legacy code. Don't feel strongly about it though.